test/goaway_test.go

```diff
@@ -569,18 +570,42 @@ func (s) TestGoAwayThenClose(t *testing.T) {
 			}
 			return err
 		},
+		S: grpc.NewServer(),
 	}
-	testgrpc.RegisterTestServiceServer(s1, ts)
-	go s1.Serve(lis1)
+	stubserver.StartTestService(t, ss1)
+	defer ss1.S.Stop()
+	//testgrpc.RegisterTestServiceServer(s1, ts)
+	//go s1.Serve(lis1)
 
 	conn2Established := grpcsync.NewEvent()
 	lis2, err := listenWithNotifyingListener("tcp", "localhost:0", conn2Established)
 	if err != nil {
 		t.Fatalf("Error while listening. Err: %v", err)
 	}
-	s2 := grpc.NewServer()
+	/*s2 := grpc.NewServer()
 	defer s2.Stop()
-	testgrpc.RegisterTestServiceServer(s2, ts)
+	testgrpc.RegisterTestServiceServer(s2, ts)*/
+	ss2 := &stubserver.StubServer{
+		Listener: lis2,
+		UnaryCallF: func(context.Context, *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+			return &testpb.SimpleResponse{}, nil
+		},
+		FullDuplexCallF: func(stream testgrpc.TestService_FullDuplexCallServer) error {
+			if err := stream.Send(&testpb.StreamingOutputCallResponse{}); err != nil {
+				t.Errorf("unexpected error from send: %v", err)
+				return err
+			}
+			// Wait forever.
```

Comment:
Its not wait forever. Its wait until a message is received from client

Reply:
Done